packages/firestore/src/core/sync_engine.ts

```diff
@@ -925,28 +946,13 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     const activeQueries: QueryData[] = [];
     const newViewSnapshots: ViewSnapshot[] = [];
     for (const targetId of targets) {
-      let queryData: QueryData;
-      const queryView = this.queryViewsByTarget[targetId];
-      if (queryView) {
-        // For queries that have a local View, we need to update their state
-        // in LocalStore (as the resume token and the snapshot version
-        // might have changed) and reconcile their views with the persisted
-        // state (the list of syncedDocuments may have gotten out of sync).
-        await this.localStore.releaseQuery(
-          queryView.query,
-          /*keepPersistedQueryData=*/ true
-        );
-        queryData = await this.localStore.allocateQuery(queryView.query);
-        const viewChange = await this.synchronizeViewAndComputeSnapshot(
-          queryView
-        );
-        if (viewChange.snapshot) {
-          newViewSnapshots.push(viewChange.snapshot);
-        }
-      } else {
+      let queryData: QueryData | undefined = undefined;
+      const queries = this.queriesByTarget[targetId];
+
+      if (!queries || queries.isEmpty()) {
```

Comment:
In general, a smaller diff is better and make this easier to review. Do you mind flipping the if-statement around again?